lib/cmds/space_cmds/generate_cmds/migration.js

```diff
@@ -0,0 +1,237 @@
+    b.memberExpression(
+      base,
+      b.identifier(identifier)
+    ),
+    [toAst(value)]
+  ), rest)
+}
+
+const createContentType = function (ct) {
+  const id = ct.sys.id
+  const { name, description, displayField } = ct
+  const chain = [
+    ['name', name],
+    ['description', description],
+    ['displayField', displayField]
+  ].filter(([id, value]) => value !== null)
+
+  const variableName = ctVariableEscape(id)
+
+  const createCallExpression = b.callExpression(
+    b.memberExpression(
+      b.identifier('migration'),
+      b.identifier('createContentType')
+    ),
+    [b.literal(id)]
+  )
+
+  const callChain = createCallChain(createCallExpression, chain)
+
+  const withDeclaration = b.variableDeclaration(
+    'const',
+    [
+      b.variableDeclarator(
+        b.identifier(variableName),
+        callChain
+      )
+    ]
+  )
+
+  return withDeclaration
+}
+
+const createField = function (ctId, field) {
+  const fieldId = field.id
+  const ctVariable = ctVariableEscape(ctId)
+
+  const chain = Object.keys(_.omit(field, 'id')).map((key) => {
+    return [key, field[key]]
+  })
+
+  return createCallChain(
+    b.callExpression(
+      b.memberExpression(
+        b.identifier(ctVariable),
+        b.identifier('createField')
+      ),
+      [b.literal(fieldId)]
+    ),
+    chain
+  )
+}
+
+const changeEditorInterface = function (ctId, fieldId, widgetId, settings) {
+  const ctVariable = ctVariableEscape(ctId)
+  settings = settings || {}
+
+  const settingsExpression = b.objectExpression(
+    _.map(settings, (v, k) => {
+      return b.property('init', b.identifier(k), b.literal(v))
+    })
+  )
+
+  return b.callExpression(
+    b.memberExpression(
+      b.identifier(ctVariable),
+      b.identifier('changeEditorInterface')
+    ),
+    [b.literal(fieldId), b.literal(widgetId), settingsExpression]
+  )
+}
+
+const getContentTypes = async function (environment, contentTypeId) {
+  return contentTypeId ? [await environment.getContentType(contentTypeId)] : (await environment.getContentTypes()).items
+}
+
+const generateContentTypeMigration = async function (environment, contentType) {
+  log(`Creating migration for content type: '${contentType.sys.id}'`)
+
+  const fieldCreators = contentType.fields.map((field) => {
+    return b.expressionStatement(createField(contentType.sys.id, field))
+  })
+
+  let editorInterfacesCreators = []
+  try {
+    log('Fetching editor interface')
+
+    const editorInterface = await environment.getEditorInterfaceForContentType(contentType.sys.id)
+    editorInterfacesCreators = contentType.fields.map((field) => {
+      const control = _.find(editorInterface.controls, (control) => {
+        return control.fieldId === field.id
+      })
+      const widgetId = control.widgetId
+      const settings = control.settings
+
+      return b.expressionStatement(changeEditorInterface(contentType.sys.id, field.id, widgetId, settings))
+    })
+  } catch (err) {
```

Comment:
Not really, if it 404s, it's because there are no fields on the CT, which is an OK state for me, maybe what I can do is check wether or not the CT actually doesn't have fields and act based on that? WDYT?

Reply:
since we are logging `Fetching editor interface` in line 176 we should log in here also to mark the end of this task.
Something like the contentType does not have any fields

Reply:
Yeah, I'm thinking just a log "The content type X does not have any fields".

Just to make sure that when someone is trying to figure out what's going on it's not just swallowed in a catch. I know it's a very rare edge case, but a simple log costs nothing, right.